Update Ethereum Comparison (#5634)
w3f/polkadot-wiki: docs/learn/learn-comparison-ethereum-2.md

slug: ../learn-comparisons-ethereum-2
---

Both protocols are blockchains at their core, but serve fundamentally different roles in how they
are utilized:

- Ethereum is a general-purpose blockchain that hosts the Ethereum Virtual Machine, an environment
  for executing smart contracts. Ethereum is homogenous, but can utilize rollups and layer two
  solutions to scale its usage.
- Polkadot is a heterogeneous, multi-chain protocol (a "layer 0", or metaprotocol) that hosts
  multiple layer 1 blockchains and provides a way for them to partake in shared security. Polkadot
  acts as a **meta-protocol** that allows for multiple protocols to coexist and work together.

> :::note

Reviewer: ```suggestion
:::note Sharding
```

The term "shards" or "sharded protocol" may be used throughout the various ecosystems. They are
typically used, in the context of blockchains, to refer to sub-protocols, or as a general term to
refer to a form of

> [**horizontal scaling**](https://en.wikipedia.org/wiki/Scalability#Horizontal_or_scale_out).

Reviewer: ```suggestion
In the context of blockchains, the term "shards" or "sharded protocol" is typically used to refer to sub-protocols or as a general term to refer to a form of
[**horizontal scaling**](https://en.wikipedia.org/wiki/Scalability#Horizontal_or_scale_out).
```

:::

## High-Level Comparison

> Both protocols have fundamentally different goals, which can be summarized as:

Reviewer: ```suggestion
Both protocols have fundamentally different goals:
```

- Ethereum is a general-purpose blockchain, which coordinates the global coordination of the

>   Ethereum Virtual Machine (EVM). Ethereum is not specialized nor optimized for any particular

Reviewer: ```suggestion
- Ethereum is a general-purpose blockchain based on the
  Ethereum Virtual Machine (EVM). Ethereum is not specialized nor optimized for any particular
```

  application, rather its primary focus is the Ethereum Virtual Machine for executing smart

>   contracts. Ethereum achieves scalability via **rollups**, which are secondary protocols which

Reviewer: ```suggestion
  contracts. Ethereum achieves scalability via [**rollups**](./learn-comparisons-rollups), which are secondary protocols which
```

  utilize Ethereum as a settlement layer.

> - Polkadot is a multi-chain protocol that provides shared security for each one of its

Reviewer: ```suggestion
- Polkadot is a multi-chain protocol that provides shared security and secure interoperability for each one of its
```

  **parachains**. Each parachain (which could also be called an "appchain" in this context) is
  specialized towards a specific focus and optimized towards that goal. Parachains must abide by the

>   [Parachains Protocol](./learn-parachains-protocol.md)

Reviewer: ```suggestion
  [Parachains Protocol](./learn-parachains-protocol.md).
```

Polkadot _does not_ directly run a virtual machine for smart contracts, as Polkadot's main purpose
is to validate the protocols that operate under it.

However, several parachains provide
[smart contract functionality](https://wiki.polkadot.network/docs/build-smart-contracts#parachains).
Parachains on Polkadot can even run an EVM for executing smart contracts written in Solidity using
[Frontier, an Ethereum compatibility layer for Substrate](https://github.com/polkadot-evm/frontier).

> > As a general summary, one could also say that Polkadot coordinates protocols (and their subsequent

Reviewer: I really like this! Maybe we expand a little and then finish with that sentence. We are not mentioning WASM. We could say that Ethereum is just all EVM (even if it is sharded) while Polkadot shards can have different logic as long as they compile to WASM. Does this make sense?

> validation), while Ethereum coordinates inputs and outputs for the EVM.

## Scalability Approaches

Ethereum favors a _rollup-centric_ approach for scaling transaction throughput.
[**Danksharding**](https://ethereum.org/en/roadmap/danksharding/) is how Ethereum plans to better
accommodate and facilitate rollup activity by providing better utilities, such as data availability

> via [Proto-Danksharding](https://ethereum.org/roadmap/danksharding),for rollups to record state to

Reviewer: ```suggestion
via [Proto-Danksharding](https://ethereum.org/roadmap/danksharding), for rollups to record state to
```

Ethereum.

Danksharding will allow for much more space to be utilized per block on Ethereum, where blobs of
data will be verifiable for an amount of time before being pruned from the network. This approach
will enable data availability at layer one and further enable layer two protocols on Ethereum to
flourish more readily.

In contrast, {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} require parachains to
register themselves in accordance to the [Parachains Protocol](./learn-parachains-protocol.md). Once
registered, {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} validates the state

> transitions of each parachain as per their PVF, or parachain validation function. Data availability

Reviewer: ```suggestion
transitions of each parachain as per their parachain validation function (PVF). [Data availability](./learn-parachains-protocol#availability-and-unavailability-phase)
```

relay chain aggregates information from each parachain, where validators agree upon consensus and
finality. In essence, one can look at Polkadot as a series of **runtimes**, which are state
transition functions used to describe parachains, as well as Polkadot itself.

### Forks, Upgrades, and Governance

Ethereum's governance is done off-chain, where various stakeholders come to a consensus through some
medium other than the protocol itself. Upgrades on Ethereum will follow the standard hard-fork
procedure, coordinating both the community and validators to upgrade their nodes to implement
protocol changes.

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} uses on-chain
[governance, called OpenGov](./learn-polkadot-opengov.md), to facilitate upgrades to the Polkadot
runtime. These upgrades are voted on via the stakeholders of Polkadot, where if successful, the
upgrade is enacted automatically in the blocks to come. The only time Polkadot validator operators
upgrade their nodes is when the client itself gets updated.

Because of this mechanism, {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} can enact

> chain upgrades and successful proposals using the Wasm meta-protocol _without_ a hard fork. Anything

Reviewer: I think Bill described this really well in its ETH Denver speech. The runtime is stored on chain -> this allows forkless upgrades because it is not stored in the node client. Maybe we can expand a little with a few more sentences.

Author: fixed - let me know how it is!

[GHOST](https://www.geeksforgeeks.org/what-is-ghost-protocol-for-ethereum/)-based and can both
finalize batches of blocks in one round.

For block production, both protocols use slot-based protocols that randomly assign validators to a
slot and provide a fork choice rule for unfinalized blocks. Polkadot uses
[BABE](./learn-consensus.md#badass-babe-sassafras) for block production. BABE includes two
mechanisms for selecting block producers, one of which is a fallback in case the first fails, which
allows for chain liveness. BABE produces unfinalized blocks on top of the chain already finalized by
GRANDPA.

There are two main differences between Ethereum and
{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} consensus:

1. {{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} finality protocol, GRANDPA,
   finalizes batches of blocks based on
   [availability and validity checks](./learn-parachains-protocol.md#availability-and-unavailability-phase)
   that happen as the proposed chain grows. The time to finality varies with the number of checks
   that need to be performed (and invalidity reports cause the protocol to require extra checks).

>    The expected time to finality is 12-60 seconds.

Reviewer: isn't it 30 seconds? We are mentioning this is the async backing comparison table. We should be consistent.

   guarantees while {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} can provide
   stronger guarantees with fewer validators per round.
   {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} achieves this by making validators
   distribute an [erasure coding](./learn-parachains-protocol.md#erasure-codes) to all validators in
   the system, such that anyone - not only the round's validators - can reconstruct a parachain's
   block and test its validity. This data availability is a core part of Polkadot - ensuring state
   is valid for its state transitions. The random parachain-validator assignments and secondary
   checks are performed by randomly selected validators making it less likely for the small set of
   validators on each parachain to collude.

### Staking Mechanics: Ethereum PoS vs. Polkadot NPoS

Ethereum is a proof-of-stake network that requires 32 ETH to stake for each validator instance.
Validators run a primary Beacon Chain node and multiple validator clients - one for each 32 ETH.
These validators get assigned to "committees," randomly selected groups to validate blocks in the
network.

> {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} can provide strong finality and

Reviewer: I would delete the staking paragraph and add the following:

Polkadot uses [Nominated Proof of Stake (NPoS)](./learn-phragmen) to select validators from a smaller set, letting smaller holders nominate validators to run the network while claiming the system’s rewards without running a node. Polkadot needs about five validators for each parachain in the network. For more information, see the [staking page](./learn-staking).

Author: this is great!! exactly what I was really looking for, thanks!

### Interoperability and Message Passing

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} uses
[Cross-Consensus Messaging (XCM)](./learn-xcm.md) for parachains to send arbitrary messages to each
other. Parachains open connections with each other and can send messages via their established
channels. Given that [collators](./learn-collator.md) will need to be full nodes of the relay chain
as well, they will be connected and can relay messages from parachain A to parachain B.

Messages do not pass through the relay chain. Only validity proofs and channel operations do (open,
close, etc.). This enhances scalability by keeping data on the edges of the system.

Currently, Ethereum rollups can communicate using
[shared sequencers](https://medium.com/@richardchen_81235/intro-to-shared-sequencing-1622d1fd51c9),
which provide a common grounds of interoperability between layer two solutions.

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} will add a protocol called

> [SPREE](learn-spree.md) that provides shared logic for cross-consensus messages. Messages sent with

Reviewer: Maybe here we can just mention Accords and then link to the page https://wiki.polkadot.network/docs/polkadot-direction#xcm-and-accords that also mentions SPREE.

SPREE carries additional guarantees about provenance and interpretation by the receiving chain.

## DApp Support and Development

Ethereum supports smart contract development using Solidity. These contracts are immutable, and
cannot be changed once published on-chain.

{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} supports smart contracts through
parachains, usually using the [ink! smart contract language](https://use.ink/), but also Solidity
through Frontier-enabled parachains. On Ethereum, smart contracts can call each other; however, they
are fixed on-chain to the domain of Ethereum. On Polkadot, smart contracts can call each other in
the same parachain _and_ across parachains.

On Polkadot, developers have the option of either using smart contracts, calling extrinsics from
pallets that modify the chain's state in some particular way or merely use Polkadot's RPC to
directly retrieve and act on-chain information. DApps on Polkadot are often composed of these
multiple components working together to modify, retrieve, and watch state changes live as they
happen.

For a more comprehensive list of how to build on Polkadot, be sure to check the

> ["Build" section of the wiki.](../build/build-guide.md).

Reviewer: ```suggestion
[Build Section](../build/build-guide.md).
```

## Conclusion

Ethereum and {{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} both use a sharded
model. Danksharding plans to utilize a rollup-centric approach by focusing on data availability. The
{{ polkadot: Polkadot :polkadot }}{{ kusama: Kusama :kusama }} ecosystem is secured by a main chain,
called the "relay chain," which in turn manages cores and allows tasks, such as parachains to be run

> on top of them.

Reviewer: ```suggestion
on top of those cores and messages to be sent between them.
```
